programs/eosiod/main.cpp

```diff
@@ -31,13 +31,47 @@ namespace fc {
    std::unordered_map<std::string,appender::ptr>& get_appender_map();
 }
 
+void logging_conf_loop()
+{
+  std::shared_ptr<boost::asio::signal_set> sighup_set(new boost::asio::signal_set(app().get_io_service(), SIGHUP));
+  sighup_set->async_wait([sighup_set](const boost::system::error_code& err, int /*num*/) {
+    if(!err)
+    {
+      ilog("Received HUP.  Reloading logging configuration.");
+      auto config_path = app().get_logging_conf();
+      if(fc::exists(config_path))
+      {
+        try {
+          fc::configure_logging(config_path);
+        } catch (const fc::exception& e) {
+          elog("Error reloading logging.json");
```

Comment:
can we refactor all of the catch-and-log logic into a static function called in a `catch (...)` block that rethrows the exception and translates the type-specific error after printing out the common information of "Error reloading logging.json" ; uses a nested block to deduplicate those messages; OR anything that effectively removes the long-tail of repetitive error handling from what we did `fc::configure_logging(config_path)` from what we need to do next: re-initialize the appenders and re-register the handler.